channeldb/meta_test.go

```diff
@@ -0,0 +1,357 @@
+package channeldb
+
+import (
+	"bytes"
+	"github.com/boltdb/bolt"
+	"github.com/go-errors/errors"
+	"testing"
+)
+
+// TestVersionFetchPut checks the propernces of fetch/put methods
+// and also initialization of meta data in case if don't have any in
+// database.
+func TestVersionFetchPut(t *testing.T) {
+	db, cleanUp, err := makeTestDB()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanUp()
+
+	meta, err := db.FetchMeta(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if meta.dbVersionNumber != getLatestDBVersion(DBVersions) {
+		t.Fatal("initialization of meta information wasn't performed")
+	}
+
+	var newVersion uint32 = getLatestDBVersion(DBVersions) + 1
+	meta.dbVersionNumber = newVersion
+
+	if err := db.PutMeta(meta, nil); err != nil {
+		t.Fatalf("update of meta failed %v", err)
+	}
+
+	meta, err = db.FetchMeta(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if meta.dbVersionNumber != newVersion {
+		t.Fatal("update of meta information wasn't performed")
+	}
+}
+
+// TestGlobalVersionList checks that migrations are applied in proper order.
+func TestOrderOfMigrations(t *testing.T) {
+	appliedMigration := -1
+	versions := []version{
+		{0, nil},
+		{1, nil},
+		{2, func(tx *bolt.Tx) error {
+			appliedMigration = 2
+			return nil
+		}},
+		{3, func(tx *bolt.Tx) error {
+			appliedMigration = 3
+			return nil
+		}},
+	}
+
+	migrations := getMigrationsToApply(versions, 1)
+
+	if len(migrations) != 2 {
+		t.Fatal("incorrect number of migrations to apply")
+	}
+
+	if migrations[0](nil); appliedMigration == 2 {
```

Comment:
It's not clear to me what these if-statements are actually doing. 

It seems that they should be replaced with: 
```go
if migrations[0].number != 2 {
  ... 
} 

if migrations[1].number != 3 {
  ... 
} 
```